In [1]:
a=10
b=20
print(a+b)

30


In [21]:
# input()
a=input()
print(a)

 o


o


In [23]:
name=input("Enter the name:")
age=input("Enter the Age:")
gender=input("Enter the gender:")
contact=input("Enter the contact:")
address=input("Enter the adress:")
personal_detail={"name":name,"age":age,"gender":gender,"contact":contact,"address":address}


Enter the name: pihu
Enter the Age: 17
Enter the gender: f
Enter the contact: 93106
Enter the adress: rohini


In [6]:
a=input("enter the  first no. :")
b=input("enter the second no. :")
print(a+b)    #string concatination

enter the  first no. : 202
enter the second no. : 145


202145


In [9]:
x="pihu"
y="bhalla"
x+y

'pihubhalla'

# type casting

  1.implicit
  2.explicit

In [10]:
a=20     #int
b=2.5    #float
a+b

22.5

In [11]:
a-b

17.5

 20.0
 -2.5
  17.5

# explicit
1. int()
2. float()
3. bool()
4. complex()
5. str()
6. list()
7. tuple()
8. set()
9. dict()
10. bin()
11. oct()
12. hex()


In [13]:
a=3.6
int(a)

3

In [14]:
s="123"
print(type(s))
x=int(s)

<class 'str'>


In [15]:
type(x)

int

In [17]:
print(x)

123


***ASSIGNMENTS***

In [1]:
# Explicit 

In [5]:
# int()- It converts a value into integer
a=3.7
print(int(a))
# float()-It converts a value into decimal number
b=5
print(float(b))
# bool()- It converts a value into true or false
print(bool(0))
print(bool(5))
#complex-It converts a value into complex form having real and imaginary part
print(complex(2,3))
#str()-changes a value into text format
c=10
print(str(c))
#list()-changes a sequence into list form using square brackets
d=(1,2,3)
print(list(d))
#tuple()-changes a sequence into tuple form using round brackets
e=[6,7,8]
print(tuple(e))
#set()-changes a sequence into set and removes repeated values uses curly brackets
f=[3,4,4,5]
print(set(f))
#dict()-changes data into key and value pair form
g={"name":"pihu","contact":2345,"age":20,'address':'Noida'}
print(g)
#bin()-changes integer into binary representation
print(bin(5))
#oct()-changes integer into octal representation 
print(oct(8))
#hex()-changes integer into hexadecimal representation
print(hex(10))

3
5.0
False
True
(2+3j)
10
[1, 2, 3]
(6, 7, 8)
{3, 4, 5}
{'name': 'pihu', 'contact': 2345, 'age': 20, 'address': 'Noida'}
0b101
0o10
0xa


#Difference between Implicit and Explicit

#Example

In [9]:
x=14
x

14

In [5]:
x=14
print(x)

14


#Datatypes

In [10]:
a=8
b=-9
c=7.6509
d="hello"
e=True
f=0
print(type(a))
print(type(b))
print(type(c))
print(type(d))
print(type(e))
print(type(f))


<class 'int'>
<class 'int'>
<class 'float'>
<class 'str'>
<class 'bool'>
<class 'int'>


In [2]:
A=43

In [3]:
bin(A)

'0b101011'

In [4]:
bin(97)

'0b1100001'

In [5]:
bin(259)

'0b100000011'

In [7]:
 bin(361)

'0b101101001'

In [9]:
0b11001110

206

In [10]:
int("11001110",2)

206

In [1]:
oct(231)

'0o347'

In [2]:
oct(423)

'0o647'

In [3]:
bin(569)

'0b1000111001'

In [4]:
0b1011100111

743

In [6]:
x=24
print("binary:",bin(x))

binary: 0b11000


In [7]:
bin(24)

'0b11000'

In [8]:
oct(24)

'0o30'

In [12]:
x=24
print("octal:",oct(x))

octal: 0o30


In [1]:
x=24
print("hexadecimal:",hex(x))

hexadecimal: 0x18


In [9]:
hex(24)

'0x18'

In [2]:
print(10+20)

30
